core/src/main/resources/jenkins/security/apitoken/LegacyApiTokenAdministrativeMonitor/resources.js

```diff
@@ -0,0 +1,150 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2018, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+function selectAll(anchor){
+    var parent = anchor.up('.legacy-token-usage');
+    var allCheckBoxes = parent.querySelectorAll('.token-to-revoke');
+    var concernedCheckBoxes = allCheckBoxes;
+    
+    checkTheDesiredOne(allCheckBoxes, concernedCheckBoxes);
+}
+
+function selectFresh(anchor){
+    var parent = anchor.up('.legacy-token-usage');
+    var allCheckBoxes = parent.querySelectorAll('.token-to-revoke');
+    var concernedCheckBoxes = parent.querySelectorAll('.token-to-revoke.fresh-token');
+    
+    checkTheDesiredOne(allCheckBoxes, concernedCheckBoxes);
+}
+
+function selectRecent(anchor){
+    var parent = anchor.up('.legacy-token-usage');
+    var allCheckBoxes = parent.querySelectorAll('.token-to-revoke');
+    var concernedCheckBoxes = parent.querySelectorAll('.token-to-revoke.recent-token');
+    
+    checkTheDesiredOne(allCheckBoxes, concernedCheckBoxes);
+}
+
+function checkTheDesiredOne(allCheckBoxes, concernedCheckBoxes){
+    var mustCheck = false;
+    for(var i = 0; i < concernedCheckBoxes.length ; i++){
+        var checkBox = concernedCheckBoxes[i];
+        if(!checkBox.checked){
+            mustCheck = true;
+        }
+    }
+    
+    for(var i = 0; i < allCheckBoxes.length ; i++){
```

Comment:
Maybe forEach() to simplify code?

Reply:
I am not a huge fan of anonymous function and I do not want to introduce es6 arrows there. We gain 5 lines with your proposal, seems ok.

Reply:
But seems to make the tests to fail, so I revert